File: backend/backend/src/WebSocketComponent/Common/DTOs/TestMessage.cs
using System.Text.Json.Serialization;
using backend.WebSocketComponent.Common.DTOs;

namespace backend.WebSocketComponent.Common.DTOs;

public class TestMessage : ITypedMessage
{
    public static string TypeString => "test";

    [JsonIgnore]
    public string InstanceTypeString => TypeString;
    
    [JsonPropertyName("message")]
    public required string Message { get; set; }
}

File: frontend/src/types/common/TestMessage.ts
import { ITypedMessage } from "../ITypedMessage";
import { MessageType } from "../MessageType";

export class TestMessage implements ITypedMessage {
    public readonly type = MessageType.TEST;
    public msg: {
        message: string;
    };

    public constructor(msg: { message: string }) {
        this.msg = msg;
    }
}
